### DISCUSSION TITLE
Authorization issue with PGSQL

### DISCUSSION THREAD

**Discussion by DRNV456** [emqx/emqx#11748] [Q&A]
Dear Friends,
I have setup and configured emqx in local machine.
Implemented authentication using the PGSQL and it's working completely fine.
When I am trying to implement Authorization using PGSQL it's allowing client to subscribe/publish to any topic.
Although I have specified rule in table that user shouldn't be allowed to subscribe/publish any topic other than specified in table in PGSQL, it's allowed to subscribe to any topic.

This is what I have stored in table in PGSQL.
id  ------------->1
ipaddress------>empty
username------>sample
clientid--------->empty
action---------->all
permission----->allow
topic----------->mgb/user1/# 
qos------------->null
retain---------->null

It's allowing device to subscribe other topic as well, like if device try to subscribe/publish mgb/user2/# , then it's allowed to do so.
How can I restrict my device to subscribe/publish to any specific topic.

**Comment by keynslug** [comment]
Hi @DRNV456, it's hard to answer this without additional details. Namely: how's the pgsql authorization configured? In the meantime, you could:
1. Double check that [`authorization.no_match`](https://www.emqx.io/docs/en/v5.0/configuration/configuration-manual.html#authentication) is set to `deny` in your configuration.
2. Enable [`info` level logging](https://www.emqx.io/docs/en/v5.0/configuration/logs.html#output-logs-as-a-file) and repeat the tests, authorization-related messages should be helpful in understanding what component decides to allow operations that you expected would be denied.
